README.md
## XS-UI - 溪水组件库

<p >Xs Ui - A Vue.js 3 UI library</p>

- 💪 Written in JavaScript
- 💪 Vue 3 Composition API

### 开始

- [组件库文档](https://zhuiyue132.github.io/component-docs)

## 使用 pnpm

```bash
npm install pnpm -g
pnpm run dev # or build task and other...
```

> 注意：npm 不支持 workspace 的定义方式，所以会报错。请切到 pnpm

## 如何发布 npm 包

1. 确保本地工作区和暂存区没有未提交的内容；
1. 打开终端，执行 `node tag`, 选择你需要发布的版本号；
   1. 值得注意的是，本步骤会在远端仓库创建一个以`v{version}`的 tag。
   1. 每种版本号有不同含义，建议慎重选择。
   1. tag 创建后，gitee go 会以正则捕获这个 git tag 的 hook，默认会执行流水线任务。
   1. 流水线内的任务定义在 `.workflow`内。
   1. 本质上执行的是 `node release.js`。
   1. 如果需要修改包的所有者，请修改 `NPM_PUBLISH_TOKEN`。

## 如何发布文档站点

- 跟随上述发 npm 流程自动更新文档站点;
- 手动发布比较繁琐，等我写个脚本简化操作后再更新文档;


CHANGELOG.md
## [1.0.14](https://gitee.com/creektech/xishui-ui/compare/v1.0.14-beta.9...v1.0.14) (2023-02-14)

### Features

- **goods:** 增加 id 复制成功的提示; ([c2be988](https://gitee.com/creektech/xishui-ui/commits/c2be988abf2fe1fe63106d001cb56f571f21151e))

## [1.0.14-beta.9](https://gitee.com/creektech/xishui-ui/compare/v1.0.14-beta.8...v1.0.14-beta.9) (2023-02-13)

### Bug Fixes

- **goods:** 修正商品主图的 popper 样式丢失的问题; ([b6fd0b1](https://gitee.com/creektech/xishui-ui/commits/b6fd0b13869c816e2e1376e48b45b77927224003))

## [1.0.14-beta.8](https://gitee.com/creektech/xishui-ui/compare/v1.0.14-beta.7...v1.0.14-beta.8) (2023-02-13)

### Bug Fixes

- 修正一些组件按需加载时应该使用 scss 却引入了 css 的问题; ([2cc7763](https://gitee.com/creektech/xishui-ui/commits/2cc7763d59ef84c9977dc7ff85be589a166bbb79))

## [1.0.14-beta.7](https://gitee.com/creektech/xishui-ui/compare/v1.0.14-beta.6...v1.0.14-beta.7) (2023-02-13)

## [1.0.14-beta.6](https://gitee.com/creektech/xishui-ui/compare/v1.0.14-beta.5...v1.0.14-beta.6) (2023-02-13)

## [1.0.14-beta.5](https://gitee.com/creektech/xishui-ui/compare/v1.0.14-beta.4...v1.0.14-beta.5) (2023-02-13)

## [1.0.14-beta.4](https://gitee.com/creektech/xishui-ui/compare/v1.0.14-beta.3...v1.0.14-beta.4) (2023-02-13)

## [1.0.14-beta.3](https://gitee.com/creektech/xishui-ui/compare/v1.0.14-beta.2...v1.0.14-beta.3) (2023-02-10)

### Bug Fixes

- **table:** 吸顶状态切换时,更新一下滚动状态,避免样式丢失; ([9a67e0d](https://gitee.com/creektech/xishui-ui/commits/9a67e0da77dfa1d672da65018a682c45cd705cbd))

## [1.0.14-beta.2](https://gitee.com/creektech/xishui-ui/compare/v1.0.14-beta.1...v1.0.14-beta.2) (2023-02-10)

### Bug Fixes

- **table:** 修正表头处于吸顶时改变窗口大小导致计算宽度错的问题; ([c204008](https://gitee.com/creektech/xishui-ui/commits/c204008a74069d947ada4434a5b2611f318917fc))

## [1.0.14-beta.1](https://gitee.com/creektech/xishui-ui/compare/v1.0.14-beta.0...v1.0.14-beta.1) (2023-02-10)

### Bug Fixes

- 组件按需加载的样式补充; ([deb9361](https://gitee.com/creektech/xishui-ui/commits/deb9361c0ad981421146f288a3da58c01c753328))

## [1.0.14-beta.0](https://gitee.com/creektech/xishui-ui/compare/v1.0.13...v1.0.14-beta.0) (2023-02-10)

### Features

- 增加 tooltip 组件和维护 goods 组件; ([1514f3e](https://gitee.com/creektech/xishui-ui/commits/1514f3e81735ab64da3f583e0e8569daeb0e3b4f))

## [1.0.13](https://gitee.com/creektech/xishui-ui/compare/v1.0.12...v1.0.13) (2023-02-10)

### Features

- 增加 tooltip 组件和维护 goods 组件; ([0f8d8ab](https://gitee.com/creektech/xishui-ui/commits/0f8d8abb0805bc905a38a82edf7b7603ab6c1a41))
- **table:** 表头吸顶样式改版;表头自定义 tooltip; ([09be532](https://gitee.com/creektech/xishui-ui/commits/09be5329d32b0bcbb07dc45f34a0b05aa936fa2d))
- **table:** 增加表格单选的支持; ([02ed2c8](https://gitee.com/creektech/xishui-ui/commits/02ed2c85ead63511e1ec2bf2d4c0a84cae268221))

## [1.0.12](https://gitee.com/creektech/xishui-ui/compare/v1.0.11...v1.0.12) (2023-02-08)

### Bug Fixes

- 修正鼠标事件对象缺失了 path 属性导致筛选功能异常的问题;一些格式化问题的修改; ([6c3533e](https://gitee.com/creektech/xishui-ui/commits/6c3533e08917d8c485886e1d103d2d9a0e07c6f8))
- **table:** 解决打包过程中出现循环依赖的警告; ([c0ffb41](https://gitee.com/creektech/xishui-ui/commits/c0ffb4161c995049a3fd7678e7a903e7aceb77d2))

### Features

- 增加样式覆盖 xishui-ui 时,自动透传到 element;增加样式主题的文档; ([744279a](https://gitee.com/creektech/xishui-ui/commits/744279a79cd93b5c79d8872c7eabc52adcaeed0f))
- **table:** select 筛选类型兼容带有英文逗号的数据生成 option; ([049f1c8](https://gitee.com/creektech/xishui-ui/commits/049f1c8be5ad1526ea47091d41f3f95182af1202))

## [1.0.11](https://gitee.com/creektech/xishui-ui/compare/v1.0.10...v1.0.11) (2023-02-07)

## [1.0.10](https://gitee.com/creektech/xishui-ui/compare/v1.0.9...v1.0.10) (2023-02-07)

### Bug Fixes

- 删除不合规范的组件; ([0c6cc04](https://gitee.com/creektech/xishui-ui/commits/0c6cc04ba17212f3501c2739c94e91cc40d4e0fd))

## [1.0.9](https://gitee.com/creektech/xishui-ui/compare/v1.0.8...v1.0.9) (2023-02-07)

### Features

- **table:** 增加表格的按需加载; ([c942418](https://gitee.com/creektech/xishui-ui/commits/c942418d77d1aea73bd037d31a730801c26e7709))

## [1.0.8](https://gitee.com/creektech/xishui-ui/compare/v1.0.7...v1.0.8) (2023-02-07)

### Bug Fixes

- **table:** 修正表格分页 size-change 和 current-change 回调执行两次的问题; ([32df40a](https://gitee.com/creektech/xishui-ui/commits/32df40ad00cc934bc8f62170d440d7e51daec3dc))

### Features

- **table:** 薪资,警告,趋势三个数据形态的封装; ([6f4143f](https://gitee.com/creektech/xishui-ui/commits/6f4143f408ef9406d3a544203bb50923eb3658ae))
- **table:** 增加了一些单元格数据形态的封装; ([1395029](https://gitee.com/creektech/xishui-ui/commits/1395029bb43bc0c8d4e5dabb4a40dd59c87e9365))

## [1.0.7](https://gitee.com/creektech/xishui-ui/compare/v1.0.7-beta.12...v1.0.7) (2023-01-14)

### Bug Fixes

- 解决因服务端没有 document,window 等对象的报错的问题; ([c1701c6](https://gitee.com/creektech/xishui-ui/commits/c1701c64834a41c8b18b9d85a9039232139a7b9a))

### Features

- 增加 Table 组件及其文档; ([ba9202c](https://gitee.com/creektech/xishui-ui/commits/ba9202c5f8eb1973d398cc26aa06dd5356ee4933))

### Reverts

- Revert "docs(release): auto update" ([4452ff0](https://gitee.com/creektech/xishui-ui/commits/4452ff0f357fbf6701cf2a4674d3baaf3da6c24b))

## [1.0.7-beta.12](https://gitee.com/creektech/xishui-ui/compare/v1.0.7-beta.11...v1.0.7-beta.12) (2023-01-06)

### Bug Fixes

- **tabs:** 解决 tab-pane 样式丢失问题; ([0bebef4](https://gitee.com/creektech/xishui-ui/commits/0bebef43baf5b0b0e9627e11a85bce8a173ad6a6))

## [1.0.7-beta.11](https://gitee.com/creektech/xishui-ui/compare/v1.0.7-beta.10...v1.0.7-beta.11) (2023-01-05)

### Bug Fixes

- **empty:** 调整 Empty 组件图片的导入方式; ([b9159bc](https://gitee.com/creektech/xishui-ui/commits/b9159bc8b15bbe5847599226da3f457d65985704))

## [1.0.7-beta.10](https://gitee.com/creektech/xishui-ui/compare/v1.0.7-beta.9...v1.0.7-beta.10) (2023-01-05)

### Bug Fixes

- 调整 MessageBox 图片导入方式; ([9cab9be](https://gitee.com/creektech/xishui-ui/commits/9cab9beeaeefe7de0da665df8ca260700d676279))

## [1.0.7-beta.9](https://gitee.com/creektech/xishui-ui/compare/v1.0.7-beta.8...v1.0.7-beta.9) (2023-01-05)


tag.js
/*
 * @Author: chenghao
 * @Date: 2022-12-06 16:13:41
 * @Last Modified by: chenghao
 * @Last Modified time: 2022-12-06 17:34:21
 * @Desc: 自动发布到npm;
 */
const process = require('process');
const path = require('path');
const fs = require('fs');
const semver = require('semver');
const inquirer = require('inquirer');
const execa = require('execa');
const consola = require('consola');

const root = process.cwd();
const rootPkgInfo = require(path.resolve(root, 'package.json'));
const xpPkgInfo = require(path.resolve(`${root}/packages/xishui-ui`, 'package.json'));
const currentVersion = rootPkgInfo.version;
const semverReleaseType = ['patch', 'minor', 'major', 'prepatch', 'preminor', 'premajor', 'prerelease'];

/**
 * 打印步骤
 */
const step = msg => consola.success(msg);

/**
 * 询问并更新版本号
 */
async function updateVersion() {
  step('更新版本号');
  const { targetVersion } = await inquirer.prompt([
    {
      type: 'list',
      name: 'targetVersion',
      message: '选择你想要发布的版本: ',
      choices: semverReleaseType
        .map(release => semver.inc(currentVersion, release, 'beta')) // semver.inc 方法递增版本号
        .map(v => ({ value: v, title: v }))
    }
  ]);

  // 更新版本号并写入 package.json 文件中
  rootPkgInfo.version = targetVersion;
  xpPkgInfo.version = targetVersion;
  fs.writeFileSync(path.resolve(root, 'package.json'), JSON.stringify(rootPkgInfo, null, 2) + '\n');
  fs.writeFileSync(
    path.resolve(`${root}/packages/xishui-ui`, 'package.json'),
    JSON.stringify(xpPkgInfo, null, 2) + '\n'
  );
  return targetVersion;
}
/**
 * 打 tag 并推送到远程仓库
 */
async function gitTag(targetVersion) {
  step('打 TAG');
  const suffixVersion = `v${targetVersion}`;
  await execa('git', ['tag', suffixVersion], { stdio: 'inherit' });
  await execa('git', ['push', 'origin', `refs/tags/${suffixVersion}`], { stdio: 'inherit' });
  await execa('git', ['push'], { stdio: 'inherit' });
}
/**
 * 生成 changelog 文件，同时将 changelog 及 package.json 更改提交
 */
async function generateChangelog(targetVersion) {
  step('\n生成 changelog');
  await execa('pnpm', ['changelog'], { stdio: 'inherit' });

  // commit changes
  const { stdout } = await execa('git', ['diff'], { stdio: 'pipe' });
  if (stdout) {
    // 文件有变化，提交代码
    await execa('git', ['add', '-A'], { stdio: 'inherit' });
    await execa('git', ['commit', '-m', `chore(release): publish v${targetVersion}`], { stdio: 'inherit' });
  } else {
    console.log('No changes to commit.');
  }
}

// 组合发布流程并执行
(async function main() {
  const targetVersion = await updateVersion();
  await generateChangelog(targetVersion);
  await gitTag(targetVersion);
})().catch(err => {
  throw err;
});
